include/eve/detail/meta.hpp

```diff
@@ -367,6 +368,32 @@ namespace eve::detail
   struct as_trait : as_trait_impl<Concept, types<T...>>
   {};
 
+  // Can I pass Type to a constant ?
+  template<typename Constant, typename Type>
+  struct is_generator : std::is_invocable<Constant,as<Type>>
+  {};
+
+  template<typename Constant, typename Type, typename Index>
+  struct  is_generator_impl;
+
+  template<typename Constant, typename Type, std::size_t... I>
+  struct  is_generator_impl<Constant,Type,std::index_sequence<I...>>
+        : std::conjunction<is_generator<Constant, std::tuple_element_t<I,Type>>...>
```

Comment:
just use a fold expression, it's faster to compile

Reply:
Done